### PR TITLE
feat(scheduling): add `noDeadlock` option for scheduling

### PR DESCRIPTION
**Pull request by sjelin**
Resolves https://github.com/sjelin/zoned-scheduler/issues/2

### CODE REVIEW

File: spec/zs_spec.ts [hunk 1]
```diff
@@ -75,4 +75,15 @@ describe('Naive Scheduler', () => {
...
+  describe('noDeadlock', () => {
+    it('should handle the basic case for noDeadlock', (done) => {
+      let scheduler = new Scheduler();
+      scheduler.schedule(() => {
+        return scheduler.schedule(() => {
+          done();
+        });
+      }, AsyncStyle.INFER, true);
+    });
+  });
```
Comment (juliemr) [inline spec/zs_spec.ts:88]: A thought: can we add a spec at the end of this file like:

```ts
it('should ensure there are no pending errors, (done) => {
  setTimeout(done, 200);
});
```

Comment (sjelin) [reply spec/zs_spec.ts:88]: done

**Comment by sjelin** [comment]
@juliemr all comments addressed

**Comment by juliemr** [comment]
LGTM!

**Pull request merged**